packages/firestore/src/local/local_documents_view.ts

```diff
@@ -218,29 +221,38 @@ export class LocalDocumentsView {
         );
       })
       .next(matchingMutationBatches => {
-        for (const batch of matchingMutationBatches) {
-          for (const mutation of batch.mutations) {
-            const key = mutation.key;
-            // Only process documents belonging to the collection.
-            if (!query.path.isImmediateParentOf(key.path)) {
-              continue;
-            }
-
-            const baseDoc = results.get(key);
-            const mutatedDoc = mutation.applyToLocalView(
-              baseDoc,
-              baseDoc,
-              batch.localWriteTime
-            );
-            if (mutatedDoc instanceof Document) {
-              results = results.insert(key, mutatedDoc);
-            } else {
-              results = results.remove(key);
+        mutationBatches = matchingMutationBatches;
+        // It is possible that a PatchMutation can make a document a match to the query,
+        // but the version in `remoteDocumentCache` is not a match yet (waiting for server
```

Comment:
Nit: s/`remoteDocumentCache`/the RemoteDocumentCache/ (since it is a widely used concept in Firestore)

Reply:
Done.